examples/mobile/build_rust_deps.sh

```diff
@@ -48,8 +48,8 @@ for arch in $ARCHS; do
         # Hardware iOS targets
         cargo rustc --crate-type staticlib --lib $RELFLAG --target aarch64-apple-ios
       else
-        # M1 iOS simulator -- currently in Nightly only and requires to build `libstd`
-        cargo rustc --crate-type staticlib --lib $RELFLAG --target aarch64-apple-ios-sim
+        # M1 iOS simulator
+        RUSTFLAGS=--cfg=ios_simulator cargo rustc --crate-type staticlib --lib $RELFLAG --target aarch64-apple-ios-sim 
```

Comment:
Not a fan of needing to specify this, since this breaks into the realm of relying on users' development environment configuration, which is notoriously hard to tame and account for. Is there another way around this?

Reply:
I think the other ways that could work is a manual feature to add to the mobile example, and bubble it to bevy, then bevy_internal, then bevy_render, and enable it when building for the simulator, or a build script. The only difference that I could find when building for the simulator is the target triple is not the same, but that's only exposed to build  scripts.

Reply:
imo in this particular case it's probably fine. it only affects the simulator, and it should be temporary.

might be nice to add something to the example std-output on ios to declare whether it's enabled or not, just for some extra breadcrumbs.

Reply:
@james7132 is that OK for you? Building for iOS needs custom scripts anyway

Reply:
If it's likely to be temporary and we can't avoid custom build scripts, I'm more OK with it. Just didn't want to unnecessarily complicate the build process.

Reply:
One thing to try if only the target triple changes might be to add a .cargo/config.toml line that adds the flag for that target triple.

https://doc.rust-lang.org/cargo/reference/config.html

Says here that a specific target triple can be used to specify rustc flags.

Reply:
that looks better, and it works! thanks